### ISSUE TITLE
Integrate with Recon

### ISSUE THREAD

**Issue by mxstbr** [carteb/carte-blanche#293]
@chrisui release Recon at ReactEurope a few days ago, and it's basically a much better `react-docgen`! 

We should look into replacing `docgen` with this asap: https://github.com/recon-js/recon

**Comment by jdfweb** [comment]
+1 on this too!

**Comment by Chrisui** [comment]
👍

So a quick update: I will be on holiday for the next week and a bit but after that I will be around to personally get Recon integrated into Carte Blanche!

Before that I'm hoping to put together some docs around the interfaces of Recon so that we can, as a community, ensure everyone's tool's needs are met :)

**Comment by mxstbr** [comment]
Awesome, I can't wait! The data recon puts out seems a lot more intricate than `docgen`, and it'll hopefully improve the experience for everyone. (imported PropTypes?!)

**Comment by unindented** [comment]
@chrisui any progress here? Any way I can help?

**Comment by chrisui** [comment]
Right. So sorry about the delay and silence! Had a much needed holiday extension and have some crazy things occurring at work.

I have pushed a new version of `recon-engine` (the guts behind recon) which is, I'm happy to say, working very well on our pretty extensive and intricate codebase at Lystable. I'm soon going to be reaching out to whoever I can to try and target all the edge cases there are bound to be. Note this is not a very user friendly version to begin with.

Current example output (across ~1,500 modules with ~700 components, takes 15 seconds to generate) can be found here: https://raw.githubusercontent.com/lystable/recon/master/packages/recon-engine/test-run.json which is generated from this query: https://raw.githubusercontent.com/lystable/recon/master/packages/recon-engine/src/engine/__fixtures__/basic-app/query.graphql

Highlights here are pulling out prop usage data, dependencies, dependants etc!

The current big blockers for any usage within carte-blanch I see are:
- Extracting the prop type information about components (won't be anything too difficult and as you can see from the linked output we have some pretty good power around resolving symbols and types across module boundaries so a level on top of react-docgen)
- Deciding how carte-blanche wants to interface with recon. All we are opinionated on is "there is a graphql interface". You can generate all the data up front (as tests do currently) for your components or query *later* on a per-component basis. (since large codebases get up to a few mb of data maybe the latter is preferable). All that is required is that `react-engine` is spawned somewhere so it can discover all the users components and provide an interface for querying.

Anyway just wanted to give a quick update on progress and allow any further questions to be asked!
